src/NametagModule.lua

```diff
@@ -108,6 +108,36 @@ module.linkTag = function(player, tag, groupId)
 		player:GetPropertyChangedSignal("Team"):Connect(updateTeamColor)
 	end
 
+	if config.options["grouproleColoredWithTeamColor"] == true then
+		if config.options["useDisplayName"] == true then
```

Comment:
Did you mean to check this? It look like the same thing happened whether display names are used or not. Perhaps you meant to change mainState to extraStat in the else block?

Reply:
I'm not actually sure what do you mean by the same thing, but if the display names are used then the group role will use ExtraStat instead of MainStat, or perhaps I'm understanding something wrong...

Reply:
Never mind, I missed the swap in line 114. My bad.